File: go/prime-factors/prime_factors.go
package prime

const testVersion = 2

// Factors returns the prime factors of a number.
func Factors(n int64) []int64 {
	factors := []int64{}

	f := int64(2)

	for n > 1 {
		if n%f == 0 {
			factors = append(factors, f)
			n /= f
		} else {
			f++
		}
	}

	return factors
}

File: ruby/prime-factors/prime_factors.rb
module PrimeFactors
  def self.for(number)
    factors = []
    factor = 2

    while number > 1
      if (number % factor).zero?
        factors << factor
        number /= factor
      elsif factor > 2
        factor += 2
      else
        factor += 1
      end
    end

    factors
  end
end
